python/GafferDispatch/TaskContextProcessor.py

```diff
@@ -53,7 +53,7 @@ def preTasks( self, context ) :
 		for plug in self["preTasks"] :
 
 			source = plug.source()
-			if source.isSame( plug ) :
+			if source.isSame( plug ) or not source.direction() == Gaffer.Plug.Direction.Out:
```

Comment:
This shouldn't be necessary after #3025 is merged. It was really error-prone expecting the TaskNodes to do this, and it goes against the mantra of nodes not knowing about the outside world. I've now moved all the responsibility to the dispatcher...